lib/view.js

```diff
@@ -0,0 +1,230 @@
+/** @babel */
+
+import {Emitter} from 'event-kit';
+
+import {Path} from './models';
+import {cachedProperty, closest, dom} from './utils';
+
+
+export default class AdvancedOpenFileView {
+    constructor() {
+        this.emitter = new Emitter();
+        this.cursorIndex = null;
+        this._updatingPath = false;
+
+        // Element references
+        this.pathInput = this.content.querySelector('.path-input');
+        this.pathList = this.content.querySelector('.list-group');
+        this.parentDirectoryListItem = this.content.querySelector('.parent-directory');
```

Comment:
This seems like it shouldn't be part of the view, if you're going for a strict MVC model.